### PR TITLE
Adds v2 Remote Config Triggers

### PR DESCRIPTION
**Pull request by colerogers**
Converted the [diff sample](https://github.com/firebase/functions-samples/tree/main/remote-config-diff) to use v2 functions

### CODE REVIEW

File: 2nd-gen/remote-config-diff/functions/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "remote-config-diff",
+  "description": "Get the difference between the previous and current Remote Config",
```
Comment (jhuleatt) [inline 2nd-gen/remote-config-diff/functions/package.json:3]:
```suggestion
  "description": "Get the difference between the previous and current Remote Config template",
```

**Review by jhuleatt — APPROVED** [review]

File: 2nd-gen/remote-config-diff/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+## Get the difference between the previous and current Remote Config templates.
+
+With the Firebase Remote Config Cloud Function trigger you can take action when your template has been updated.
```
Comment (jenh) [inline 2nd-gen/remote-config-diff/README.md:3]:
```suggestion
Use the Firebase Remote Config Cloud Function trigger to take action when your template is updated.
```

File: 2nd-gen/remote-config-diff/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+In this sample, the difference between the previous and current templates is logged.
```
Comment (jenh) [inline 2nd-gen/remote-config-diff/README.md:5]:
```suggestion
This sample demonstrates how to use the Remote Config Cloud Function trigger to log the difference between the previous and current Remote Config templates.
```

File: 2nd-gen/remote-config-diff/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+See [eBay's](https://github.com/eBay/firebase-remote-config-monitor) real world implementation.
```
Comment (jenh) [inline 2nd-gen/remote-config-diff/README.md:7]:
```suggestion
* See [eBay's real world implementation](https://github.com/eBay/firebase-remote-config-monitor).
```

File: 2nd-gen/remote-config-diff/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+See the [docs](https://firebase.google.com/docs/functions/rc-events) for more on Remote Config Triggers.
```
Comment (jenh) [inline 2nd-gen/remote-config-diff/README.md:9]:
```suggestion
* See the [docs](https://firebase.google.com/docs/functions/rc-events) for more on Remote Config Triggers.
```

**Review by jenh — APPROVED** [review]
Added a few nits, otherwise LGTM. :)

**Pull request merged**